fiona/fio/info.py

```diff
@@ -50,15 +56,31 @@ def env(ctx, key):
 @click.option('--bounds', 'meta_member', flag_value='bounds',
               help="Print the boundary coordinates "
                    "(left, bottom, right, top).")
+@click.option('--name', 'meta_member', flag_value='name',
+              help="Print the datasource's name.")
 @click.pass_context
-def info(ctx, input, indent, meta_member):
+def info(ctx, input, indent, meta_member, layer):
+
+    """
+    Print information about a dataset.
+
+    When working with a multi-layer dataset the first layer is used by default.
+    Use the '--layer' option to select a different layer.
+    """
+
     verbosity = (ctx.obj and ctx.obj['verbosity']) or 2
     logger = logging.getLogger('fio')
     try:
         with fiona.drivers(CPL_DEBUG=verbosity>2):
-            with fiona.open(input) as src:
+            with fiona.open(input, layer=layer) as src:
                 info = src.meta
-                info.update(bounds=src.bounds, count=len(src))
+                info.update(bounds=src.bounds, name=src.name)
+                try:
+                    info.update(count=len(src))
+                except TypeError as e:
+                    info.update(count=None)
+                    msg = "{exc} Setting 'count' to 'null'".format(exc=str(e))
+                    warnings.warn(msg, RuntimeWarning)
```

Comment:
@geowurster I think OGR considers a count of `-1` (or `None` in our case) "normal" for formats like GPX (see https://github.com/Toblerity/Fiona/pull/190), so how about `log.debug()` instead of a warning?